### PR TITLE
Fix bugs related to header management

### PR DESCRIPTION
**Pull request by Ircama**
Fixing some bugs introduced with the option to add a custom *header*.

Notice that now commands with custom header are printed (`__str__`) with 7 digits (e.g., hhhcccc like in `b'7E22121': Cruise Operation Status` needing 7E2 custom header) while all standard commands are still printed with 4 digits (cccc) like in `010C: Engine RPM`. This is to differentiate commands with custom header from cases where we have the same command with default `ECU_HEADER.ENGINE` header).

Reason for not always printing 7 digits is to keep compatibility with previous implementations (and maybe with different protocols than ISO 15765-4).

If always using 7 bits (or 4 bits without header as it was before) is preferred, I'll revise the PR.

### CODE REVIEW

**Comment by alistair23** [comment]
It looks like this breaks the Travis tests, can you fix the failure?

**Comment by Ircama** [comment]
> It looks like this breaks the Travis tests, can you fix the failure?

I double checked my code and it works with python2.7 and python3.6.

I would need some help in fixing this issue which relates to the code in https://github.com/brendan-w/python-OBD/blob/master/tests/test_OBD.py.

It looks like the [test command](https://github.com/brendan-w/python-OBD/blob/master/tests/test_OBD.py#L64) is a string type instead of [bytes](https://github.com/brendan-w/python-OBD/blob/master/obd/commands.py#L53).

This makes `return hash(self.header+self.command)` fail.

Should I correct https://github.com/brendan-w/python-OBD/blob/master/tests/test_OBD.py#L64?

**Comment by Ircama** [comment]
I meanwhile checked the [documentation](https://python-obd.readthedocs.io/en/latest/Custom%20Commands/) which says that *command type* should be *bytes* and not *string*; also the other [test program](https://github.com/brendan-w/python-OBD/blob/master/tests/test_OBDCommand.py#L23) uses bytes, so I think we would need to modify the [“toy command”](https://github.com/brendan-w/python-OBD/blob/master/tests/test_OBD.py#L64) used to test.

I’ll do that in a specific PR to allow separately checking results.

Recently, [*repr*](https://github.com/brendan-w/python-OBD/blob/master/obd/OBDCommand.py#L118) has been added to to OBDCommand by @LightSoar. I think that also this method could take care of the header, but we can better verify it in a separate PR after this one.

**Comment by alistair23** [comment]
Master has been updated, can you rebase this PR?

**Comment by Ircama** [comment]
After merging https://github.com/brendan-w/python-OBD/pull/146, the Travis tests passed.

File: obd/OBDCommand.py [hunk 1]
```diff
@@ -113,17 +114,19 @@ def __constrain_message_data(self, message):
                 logger.debug("Message was shorter than expected. Padded message: " + repr(message.data))
 
     def __str__(self):
+        if self.header != ECU_HEADER.ENGINE:
+            return "%s: %s" % (self.header+self.command, self.desc)
```
Comment (alistair23) [inline obd/OBDCommand.py]: Can you add spaces between the +

File: obd/OBDCommand.py [hunk 2]
```diff
@@ -113,17 +114,19 @@ def __constrain_message_data(self, message):
...
         return "%s: %s" % (self.command, self.desc)
     
     def __repr(self):
         return "OBDCommand(%s, %s)" % (self.name, self.command)
 
     def __hash__(self):
         # needed for using commands as keys in a dict (see async.py)
-        return hash(self.command)
+        return hash(self.header+self.command)
```
Comment (alistair23) [inline obd/OBDCommand.py]: Spaces here as well?

File: obd/OBDCommand.py (hunk 1 above)
Comment (Ircama) [reply obd/OBDCommand.py]: Yep, thanks!

File: obd/OBDCommand.py (hunk 2 above)
Comment (Ircama) [reply obd/OBDCommand.py]: ok

**Comment by Ircama** [comment]
> Looks good, just two nit picks then it is ready to merge

Done and rebased, thanks

**Pull request merged**